Libraries/LibCompress/Deflate.cpp

```diff
@@ -345,12 +346,97 @@ u32 DeflateDecompressor::decode_distance(u32 symbol)
     ASSERT_NOT_REACHED();
 }
 
-void DeflateDecompressor::decode_codes(CanonicalCode&, CanonicalCode&)
+void DeflateDecompressor::decode_codes(CanonicalCode& literal_code, Optional<CanonicalCode>& distance_code)
```

Comment:
This API feels weird to me, why is an `Optional<T>` passed by reference?
is this for optimisation purposes?
the function doesn't seem to be called in a tight loop, so returning an optional would probably work as well, if not better (RVO and all)
am I missing something?

Reply:
Essentially, this function has two return values, the code used to encode literal values which is always present and the code that is used to encode distances which is only sometimes present.

Reply:
Ah, both of those are out parameters.
I guess that's fine :shrug: